### PR TITLE
#167190462 Implement role based access control functionality

### PR DESCRIPTION
**Pull request by encodedBicoding**
## user
- create articles
- like articles
- comment on articles
- follow and unfollow other users
## moderator
- do everything a user can do
- prevent other users from creating articles and commenting on articles
- delete articles and comments
## admin
- do everything a moderator can do
- make a user a moderator
- strip a user of moderator rights (by making them a basic user)
## god
- do everything an admin can do
- make a user an admin
- strip a user of admin rights (by making them a moderator or a basic user)
- delete a user's account

#### How should this be manually tested?
- fetch this branch
- install dependencies and start server `npm install` and `npm run dev`
-use the /admin endpoint or use [Swagger](http://localhost:5040/api/v1/api-docs/#/Articles/post%20comments%20%20postId%20%20)
#### Any background context you want to provide?
All implementation was done as a result of direct project owner specification.
#### What are the relevant pivotal tracker stories?
[#167190462](https://www.pivotaltracker.com/story/show/167190462)
#### Screenshots (if appropriate)
Screenshot 2019-08-07 at 11 11 56 AM

#### Questions:
N/A

### CODE REVIEW

**Review by chuxmykel — APPROVED** [review]
LGTM

**Review by Hadeneekeh — APPROVED** [review]
LGTM

**Review by Lundii — COMMENTED** [review]
Nice work  on your implementation. I just want to ask, can an individual have more than one role?

**Review by topseySuave — APPROVED** [review]

**Review by cvjude — APPROVED** [review]
LGTM

File: server/middlewares/schema.js [hunk 1]
```diff
@@ -155,16 +158,15 @@ export const resetPasswordSchema = {
     .required()
     .regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])/)
     .error((errors) => {
-      /* istanbul ignore next */
       errors.forEach((err) => {
         if (err.type === 'string.regex.base') {
           err.message = 'password must contain at least 1 uppercase, 1 lowercase, 1 number and 1 special character';
         }
       });
-      /* istanbul ignore next */
```
Comment (Hadeneekeh) [inline server/middlewares/schema.js:164]: 👍🏽

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "node build/index.js",
-    "dev": "export DEBUG=dev && nodemon --exec babel-node server/index.js",
+    "dev": "export DEBUG=dev && NODE_ENV=production && nodemon --exec babel-node server/index.js",
```
Comment (obayomi96) [inline package.json]: I think the former dev script is the once the team uses. The `NODE_ENV=production` should not to be there.

**Review by obayomi96 — APPROVED** [review]

**Pull request merged**